wrangler-service/src/main/java/co/cask/wrangler/service/bigquery/BigQueryService.java

```diff
@@ -489,10 +506,11 @@ private boolean validateConnection(String connectionId, Connection connection,
     }
     for (String whitelistedDataset : whitelistStr.split(",")) {
       whitelistedDataset = whitelistedDataset.trim();
+      // whitelistedDataset should be of the form <project>:<dataset>, or just <dataset>
+      // if it ends with ':', the admin provided an invalid entry in the whitelist and it should be ignored.
       if (whitelistedDataset.endsWith(":")) {
         continue;
       }
-      // whitelist can contain <project>:<dataset>, or just <dataset>
       int idx = whitelistedDataset.indexOf(':');
       if (idx > 0) {
```

Comment:
What happens when the value is `:dataset`? should we remove the colon when idx == 0?

Reply:
hm good point, will add that.